SonarQube.Common/CommandLine/ArgumentDescriptor.cs

```diff
@@ -0,0 +1,78 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ArgumentDescriptor.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+
+namespace SonarQube.Common
+{
+    /// <summary>
+    /// Data class that describes a single valid command line argument - id, prefixes, multiplicity etc
+    /// </summary>
+    public class ArgumentDescriptor
```

Comment:
Extracted to a separate file and moved to Common.
I've also made the properties read-only.